### PR TITLE
feat(Puray Animes): add presence

### PR DESCRIPTION
**Pull request by Luscarvalho**
Home Page
home
Profile
profile
Anime
anime
Watch
watch

### CODE REVIEW

File: websites/P/Puray Animes/presence.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	const defaultData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname } = window.location,
+		data: PresenceData = defaultData;
+	if (document.querySelector("input[id^=headlessui]"))
+		data.details = "🔍・Pesquisando Animes";
+	else if (pathname.startsWith("/home")) {
+		data.details = "Início:";
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:15]:
```suggestion
		data.details = "Início";
```

File: websites/P/Puray Animes/presence.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		data.state = "Visualizando animes.";
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:16]: presence data state and details should not have punctuation marks. Also you cannot be "viewing animes" in the home page, if anything you'd be searching or browsing them

File: websites/P/Puray Animes/presence.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		data.buttons = [{ label: "💻・Puray Animes", url: location.href }];
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:17]: this is not allowed, please remove this button

File: websites/P/Puray Animes/presence.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	} else if (pathname.startsWith("/profile/")) {
+		const username = document.querySelector("h3").childNodes[0]?.textContent;
+		data.details = "Visualizando Perfil:";
+		data.state = `${username} - ${
+			document.querySelector("button[class*=blue] span")?.textContent
+		}`;
+		data.startTimestamp = browsingTimestamp;
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:24]: this is not needed as you already set the startTimestamp above

File: websites/P/Puray Animes/presence.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		data.buttons = [{ label: `🐒・Perfil de ${username}`, url: location.href }];
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:25]: Buttons should not repeat the data shown above
```suggestion
		data.buttons = [{ label: `Ver perfil`, url: location.href }];
```

File: websites/P/Puray Animes/presence.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	} else if (pathname.startsWith("/anime/")) {
+		const genders: string[] = [];
+		for (const _ of document.querySelectorAll("div[class^=sm]>div[class^=mb]"))
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:28]: please use better names for your variables, this is not readable

File: websites/P/Puray Animes/presence.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	} else if (pathname.startsWith("/profile/")) {
+		const username = document.querySelector("h3").childNodes[0]?.textContent;
+		data.details = "Visualizando Perfil:";
+		data.state = `${username} - ${
+			document.querySelector("button[class*=blue] span")?.textContent
+		}`;
+		data.startTimestamp = browsingTimestamp;
+		data.buttons = [{ label: `🐒・Perfil de ${username}`, url: location.href }];
+	} else if (pathname.startsWith("/anime/")) {
+		const genders: string[] = [];
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:27]: A palavra "géneros" em português tem dois significados e este é o significado errado. A variável deve se chamar "genres" se estiveres a falar do género (tipo) de show

File: websites/P/Puray Animes/presence.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+		for (const _ of document.querySelectorAll("div[class^=sm]>div[class^=mb]"))
+			genders.push(_.textContent);
+
+		data.details = document.querySelector(
+			"section div[class^=text-3xl]"
+		)?.textContent;
+		if (genders.length) data.state = genders.join(", ");
+		data.startTimestamp = browsingTimestamp;
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:35]: this is useless as you already set it above

File: websites/P/Puray Animes/presence.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+		data.buttons = [{ label: "📺・Assistir Anime", url: location.href }];
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:36]:
```suggestion
		data.buttons = [{ label: "Assistir Anime", url: location.href }];
```

File: websites/P/Puray Animes/presence.ts [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+		const episode = document.querySelector(
+			"span.text-lg.font-bold"
+		)?.textContent;
+		data.details = document.querySelector(
+			"span.text-sm.font-bold.underline"
+		)?.textContent;
+		data.state = episode;
+		data.buttons = [
+			{
+				label: `📺・Assistir EP ${episode?.match(/^\d+/g)[0]}`,
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:47]: if the state already shows the episode number you shouldn't mention it here. Also remove the emoji

File: websites/P/Puray Animes/presence.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+		];
+		const video = document.querySelector("video");
+		if (!video.paused && video.readyState >= 1) {
+			[data.startTimestamp, data.endTimestamp] = presence.getTimestamps(
+				video.currentTime,
+				video.duration
+			);
+		}
+	}
+	if (!data.details) presence.setActivity(defaultData);
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:59]:
```suggestion
	if (!data.details) presence.setActivity();
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Comment by ImRodry** [comment]
@Luscarvalho please resolve the conversations you addressed, it makes everyone's lives easier :)

**Comment by Luscarvalho** [comment]
> @Luscarvalho please resolve the conversations you addressed, it makes everyone's lives easier :)

I fixed everything that was requested and I committed

**Comment by ImRodry** [comment]
@Luscarvalho I am asking you to resolve the conversations left above to make it easier to re-review

**Comment by Luscarvalho** [comment]
> @Luscarvalho I am asking you to resolve the conversations left above to make it easier to re-review



> @Luscarvalho I am asking you to resolve the conversations left above to make it easier to re-review

ok, I just did this

File: websites/P/Puray Animes/dist/metadata.json [hunk 12]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.7",
+  "author": {
+    "name": "Luscarvalho#0101",
+    "id": "458776046184169473"
+  },
+  "service": "Puray Animes",
+  "description": {
+    "en": "Puray Animes is a site to watch anime without ads.",
+		"pt_BR": "Puray Animes é um site para ver animes sem anuncios."
```
[the remark this answers is not in the document]
Comment (ImRodry) [reply websites/P/Puray Animes/dist/metadata.json:10]: This was not addressed

File: websites/P/Puray Animes/presence.ts (hunk 6 above)
Comment (ImRodry) [reply websites/P/Puray Animes/presence.ts:24]: This was not addressed

Comment (Luscarvalho) [reply websites/P/Puray Animes/presence.ts]: > This was not addressed

I fixed and committed

File: websites/P/Puray Animes/presence.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
+const presence = new Presence({
+		clientId: "972246349917610054"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+presence.on("UpdateData", () => {
+	const { pathname } = window.location,
+		data: PresenceData = {
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:7]: for the sake of consistency, please
```suggestion
		presenceData: PresenceData = {
```

File: websites/P/Puray Animes/presence.ts [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp
+		};
+	if (document.querySelector("input[id^=headlessui]"))
+		data.details = "Pesquisando Animes";
+	else if (pathname.startsWith("/home")) {
+		data.details = "Início";
+		data.state = "Navegando";
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:15]: Personally I don't think you need the two lines here and the first one should say what you're doing
```suggestion
		data.details = "Na página principal";
```

File: websites/P/Puray Animes/presence.ts [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+	} else if (pathname.startsWith("/profile/")) {
+		data.details = "Visualizando perfil:";
+		data.state = `${
+			document.querySelector("h3").childNodes[0]?.textContent
+		} - ${document.querySelector("button[class*=blue] span")?.textContent}`;
+		data.buttons = [{ label: "Ver Perfil", url: location.href }];
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:21]:
```suggestion
		data.buttons = [{ label: "Ver perfil", url: location.href }];
```

File: websites/P/Puray Animes/presence.ts [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+		if (genres.length) data.state = genres.join(", ");
+		data.buttons = [{ label: "Assistir anime", url: location.href }];
+	} else if (pathname.startsWith("/watch/")) {
+		data.details = document.querySelector(
+			"span.text-sm.font-bold.underline"
+		)?.textContent;
+		data.state = document.querySelector("span.text-lg.font-bold")?.textContent;
+		data.buttons = [
+			{
+				label: "Assistir Episódio",
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:40]:
```suggestion
				label: "Assistir episódio",
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Comment by ImRodry** [comment]
Also you did not feel the PR template at all, so please do so

File: websites/P/Puray Animes/presence.ts (hunk 13 above)
Comment (ImRodry) [reply websites/P/Puray Animes/presence.ts:7]: Please stop resolving this without addressing the review, you need to rename every instance of the variable not just this one

Comment (Luscarvalho) [reply websites/P/Puray Animes/presence.ts]: I changed the variable name.

**Comment by Luscarvalho** [comment]
unknown
ooo
aaa

File: websites/P/Puray Animes/presence.ts [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+	} else if (pathname.startsWith("/anime/")) {
+		const genres: string[] = [];
+		for (const item of document.querySelectorAll(
+			"div[class^=sm]>div[class^=mb]"
+		))
+			genres.push(item.textContent);
+		presenceData.details = document.querySelector(
+			"section div[class^=text-3xl]"
+		)?.textContent;
+		if (genres.length) presenceData.state = genres.join(", ");
```
Comment (ImRodry) [inline websites/P/Puray Animes/presence.ts:30]: I personally don't think showing the genres in the state is relevant information, maybe you could move the anime title here and have a generic phrase in the details like "Watching anime:"

Comment (Luscarvalho) [reply websites/P/Puray Animes/presence.ts:30]: Personally I found it better this way, but in the future I can change to a more generic phrase too, no problem.

Comment (ImRodry) [reply websites/P/Puray Animes/presence.ts:30]: What do you mean in the future?

Comment (Luscarvalho) [reply websites/P/Puray Animes/presence.ts:30]: I mean that I can still change the presence according to the evolution of the site, adding more features.

Comment (Luscarvalho) [reply websites/P/Puray Animes/presence.ts]: > What do you mean in the future?

Anyway, I fixed what you asked for and removed the genres.

**Review by ImRodry — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**